### ISSUE TITLE
PSalm: Could not create cache directory

### ISSUE THREAD

**Issue by Slamdunk** [laminas/laminas-continuous-integration-action#9]
Reference: https://github.com/laminas/laminas-cache/pull/72/checks?check_run_id=2149498850

```
Running ./vendor/bin/psalm --shepherd --stats --output-format=github --no-cache
Uncaught RuntimeException: PHP Error: Could not create cache directory: /github/home/.cache/psalm/99f877ecb53c330fe5ee0da5591b3ff0311fa525 in /github/workspace/vendor/vimeo/psalm/src/Psalm/Config.php:803 in /github/workspace/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:66
```

**Comment by Slamdunk** [comment]
Another fail: https://github.com/laminas/laminas-server/pull/26/checks?check_run_id=2191440262

**Comment by Ocramius** [comment]
Are these random or reproducible consistently?

**Comment by Slamdunk** [comment]
AFAICT, consistently everywhere a `psaml.xml` is committed

**Comment by weierophinney** [comment]
This is because the user the QA command runs under is a non-privileged user. For some reason, however, the default cache directory is being resolved to the root user (likely due to 4ccd3db7c9, which is used to ensure that QA scripts have access to action environment variables). (Interestingly, cache-by-default was added to Psalm after the initial batch of repos I rolled the workflow out to, so I didn't run into it until recent (< 6 week old) versions of Psalm.)

The current workaround is to add the directive `cacheDirectory="./psalm-cache"` to the psalm.xml.dist root declaration, and then put `/.psalm-cache/` in the `.gitignore` file.

I'm not quite sure how to resolve the root issue:

- The reason I run checks as an unprivileged user is to prevent scripts that might try and modify the underlying system.
- The reason we preserve env variables is to ensure GHA env vars are passed on to the QA scripts (e.g. to allow uploading coverage reports)

We _may_ be able to address it by also passing the `-H` or `--set-home`  flag to `sudo`. I'll give that a try; if it works, I'll send a PR, but otherwise, I need some ideas.
